#!/usr/bin/env python
PACKAGE = "fieldrobot_05"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
gen.add("laser_scan_max_distance", double_t, 0, "The distance the laser looks ahead", 4, 0.1, 20)
gen.add("inliers_epsilon", double_t, 0, "The the ratio between inliers and outliers before ransac terminates", 0.1, 0, 1)
gen.add("ransac_n",  int_t, 0, "Number of ransac iterations",100,1,100000)
gen.add("box_lim_x",  double_t, 0, "Look forward maximum distance",2.3,0.01,20)
gen.add("box_lim_y",  double_t, 0, "Look side maximum distance",1.4,0.01,10)
gen.add("headland_box_min_x",  double_t, 0, "Look forward headland minimum distance",0.9,0,10)
gen.add("headland_box_max_x",  double_t, 0, "Look forward headland maximum distance",1.1,0,5)
gen.add("headland_box_lim_y",  double_t, 0, "Look side headland lim distance",1.6,0,5)
gen.add("ransac_minimum_distance_fit",  double_t, 0, "the min distance between random points selected for fitting",0.5,0.01,10)
gen.add("ransac_maximum_distance_fit",  double_t, 0, "the max distance between random points selected for fitting",10,0.01,10)
gen.add("minimum_inliers", double_t, 0, "Minimum number of inliers before ransac terminates",0.4,0,1)
exit(gen.generate(PACKAGE, "dynamic_tutorials", "lidar_row_extractor_params"))
